cli/src/commands/duplicate.rs

```diff
@@ -179,6 +199,48 @@ pub(crate) fn cmd_duplicate(
         duplicate_commits_onto_parents(tx.repo_mut(), &to_duplicate)?
     };
 
+    let mut new_descs = HashMap::new();
+
+    if let Some(template) = args.template.as_ref() {
+        {
+            let parsed = tx.parse_commit_template(ui, template)?;
+
+            for (old_id, new_commit) in &duplicated_commits {
+                let mut output = Vec::new();
+                let old_commit = tx.repo().store().get_commit(old_id)?;
+                parsed
+                    .format(&old_commit, &mut PlainTextFormatter::new(&mut output))
+                    .expect("write() to vec backed formatter should never fail");
+
+                new_descs.insert(
+                    new_commit.id().clone(),
+                    (output.into_string_lossy(), old_id.clone()),
+                );
+            }
+        }
+
+        tx.repo_mut().transform_descendants(
+            duplicated_commits
+                .values()
+                .map(|c| c.id().clone())
+                .collect(),
+            |rewriter| {
+                let old_dupe_commit_id = rewriter.old_commit().id().clone();
+                let commit_builder = rewriter.reparent();
+                if let Some((desc, old_commit_id)) = new_descs.remove(&old_dupe_commit_id) {
+                    *duplicated_commits
+                        .get_mut(&old_commit_id)
+                        .expect("duplicated_commits should contain the original commit id") =
+                        commit_builder.set_description(desc).write()?;
```

Comment:
> the main downside of this method is that the evolog looks ugly with the extra commit

Perhaps, you can fix up `predecessors` to point to the original commit.

A better option is to split `duplicate_commits()` into two parts.

Reply:
Yeah, it seems like splitting it may be the best bet. I tried to do some things with callbacks, to no avail, and I was thinking about splitting them, but then I need to think about how exactly to split the functions. I'll figure it out. :)

Reply:
Right. Since the caller knows the commits to be duplicated, we don't have to split functions. (I thought there would be something that had to be computed.)